miner/inclusion_list_building.go

```diff
@@ -0,0 +1,55 @@
+package miner
+
+import (
+	"time"
+
+	"github.com/ethereum/go-ethereum/beacon/engine"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+	"github.com/ethereum/go-ethereum/params"
+)
+
+// BuildInclusionListArgs contains the provided parameters for building inclusion list.
+type BuildInclusionListArgs struct {
+	Parent common.Hash // The parent block to build inclusion list on top
+}
+
+func (miner *Miner) BuildInclusionList(args *BuildInclusionListArgs) (engine.InclusionList, error) {
+	genParams := &generateParams{
+		timestamp:   uint64(time.Now().Unix()),
+		forceTime:   false,
+		parentHash:  args.Parent,
+		coinbase:    miner.config.PendingFeeRecipient,
+		random:      common.Hash{},
+		withdrawals: []*types.Withdrawal{},
+		beaconRoot:  nil,
+		noTxs:       false,
+	}
+	env, err := miner.prepareWork(genParams, false)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := miner.fillTransactions(nil, env); err != nil {
```

Comment:
We will fill with as many transactions as possible, but only the first 8 will actually be included in the inclusion list, right? Maybe we should limit the transactions to 8 in that case to not spend too much extra time on this

Reply:
The IL size of 8kb is the limit per proposer in the committee.  The aggregated IL that's passed to the EL for block production can be 8*16 kb max.

Reply:
FYI, we plan to switch to gas limit instead of size limit. This aligns the unit used for ILs with that used for blocks and also makes FOCIL forward compatible with the post-zkEVM world by preventing ILs from be filled with "prover-killer" transactions.